analyzers/nodejs/integration_test.go

```diff
@@ -0,0 +1,211 @@
+	if err != nil {
+		panic(err)
+	}
+
+	exitCode := m.Run()
+	defer os.Exit(exitCode)
+	// uncomment this if you need test files cleaned up locally
+	// defer cleanUp(nodeAnalyzerFixtureDir)
+}
+
+// While not testing the core functionality, this ensures that the tests have been setup correctly as needed for a prereq to run the analyzer steps
+// This test itself does not incur any overhead.
+func TestTestSetup(t *testing.T) {
+	t.Parallel()
+	assertProjectFixtureExists(t, "puppeteer")
+	// faker has no deps
+	// assertProjectFixtureExists(t, "fakerjs")
+	assertProjectFixtureExists(t, "fastify")
+	assertProjectFixtureExists(t, "nest")
+	assertProjectFixtureExists(t, "ohm")
+	assertProjectFixtureExists(t, "express")
+	assertProjectFixtureExists(t, "standard")
+	assertProjectFixtureExists(t, "sodium-encryption")
+	assertProjectFixtureExists(t, "request")
+}
+
+func TestAnalysisOutput(t *testing.T) {
+	t.Parallel()
+	for _, proj := range projects {
+		t.Run(proj.Name, func(t *testing.T) {
+			module := module.Module{
+				Dir:         filepath.Join(nodeAnalyzerFixtureDir, proj.Name),
+				Type:        pkg.NodeJS,
+				Name:        proj.Name,
+				Options:     map[string]interface{}{"allow-npm-err": true},
+				BuildTarget: filepath.Join(nodeAnalyzerFixtureDir, proj.Name, "package.json"),
+			}
+
+			analyzer, err := analyzers.New(module)
+			assert.NoError(t, err)
+
+			deps, err := analyzer.Analyze()
+			assert.NoError(t, err)
+			// faker has no deps
+			if proj.Name == "fakerjs" {
+				assert.Empty(t, deps.Direct)
+				assert.Empty(t, deps.Transitive)
+			} else {
+				assert.NotEmpty(t, deps.Direct)
+				assert.NotEmpty(t, deps.Transitive)
+			}
+		})
+	}
+}
+
+func assertProjectFixtureExists(t *testing.T, name string) {
+	exists, err := files.ExistsFolder(nodeAnalyzerFixtureDir, name)
+	assert.NoError(t, err)
+	assert.True(t, exists, name+" was not properly cloned")
+
+	exists, err = files.ExistsFolder(nodeAnalyzerFixtureDir, name, "node_modules")
+	assert.NoError(t, err)
+	assert.True(t, exists, name+" did not have its node modules installed")
+}
+
+func initializeProjects(testDir string) error {
+	var waitGroup sync.WaitGroup
+	waitGroup.Add(len(projects))
+
+	for _, project := range projects {
+		go func(proj fixtures.ProjectFixture) {
+			defer waitGroup.Done()
+
+			projectDir := filepath.Join(testDir, proj.Name)
+
+			_, errOut, err := exec.Run(exec.Cmd{
+				Name:    "npm",
+				Argv:    []string{"install", "--production"},
+				Dir:     projectDir,
+				WithEnv: proj.Env,
+				Command: "npm",
+			})
+			if err != nil {
+				println(errOut)
+				println("failed to run npm install on " + proj.Name)
+			}
+
+			// save time on local
+			ymlAlreadyExists, err := files.Exists(filepath.Join(projectDir, ".fossa.yml"))
+			if err != nil {
+				panic(err)
+			}
+			if ymlAlreadyExists {
+				return
+			}
+
+			// any key will work to prevent the "NEED KEY" error message
+			err = runfossa.Init(projectDir)
```

Comment:
The need to run `fossa init` may be general enough among integration tests to extract it outside of this package.

In the `fixtures` package, it may be helpful to have `fixtures.Add(dir string, fixtures []Fixture)` which does a clone and then `fossa init`.

Reply:
While possible, I think we may want to keep this separate for now. Once we establish a pattern it may be useful to DRY it up. 

E.g. there may be tests where we want to test `fossa init` or different config files